internal/io/http/client.go

```diff
@@ -172,16 +172,11 @@ func (cc *ClientConf) InitConf(device string, props map[string]interface{}, with
 			return fmt.Errorf("headers must be a map or a string")
 		}
 	}
-	tlsOpts, err := cert.GenTlsConfigurationOptions(props)
+	tlscfg, tlsOpts, err := cert.GenTLSConfig(props)
 	if err != nil {
 		return err
 	}
 	c.TlsConfigurationOptions = tlsOpts
```

Comment:
Looks like the c.TlsConfigurationOptions is useless. Can we just remove it and let the GenTLSConfig only return two values.